pkg/cloud/libvirt/client/domain.go

```diff
@@ -460,35 +460,43 @@ type Config struct {
 	URI string
 }
 
-func domainDefInit(domainDef *libvirtxml.Domain, name string, memory, vcpu int) error {
-	if name != "" {
-		domainDef.Name = name
+func setFirmware(input *CreateDomainInput, domainDef *libvirtxml.Domain, arch string) {
+	// for aarch64 speciffying this will automatically select the firmware and NVRAM file
+	// reference: https://libvirt.org/formatdomain.html#bios-bootloader
+	if arch == "aarch64" {
+		domainDef.OS.Firmware = "efi"
+	}
```

Comment:
I am ok with hardcoding this for aarch64, but do we want to expose this through the machine provider api so it can be set through the installer and this code can be arch agnostic?

Reply:
ok..on second thought .. probably not...this is a basic property of aarch64...better to leave it as is and keep things simple

Reply:
It must not be so basic, otherwise it would be the libvirt default ;) I agree with leaving this as is for now, as I don't think cluster-api-provider-libvirt should support arbitrary VM configurations. This is one case where it should be able to pick up the correct value needed by openshift. For x86_64, the efi switch would happen when we start using the q35 machine type.